accepted/2024/unified-vs-settings.md

```diff
@@ -0,0 +1,168 @@
+1. Display text indicating the completion of the command (or any error messages) and a close button on the modal dialog.
+
+### External Settings API
+
+To map each of our registered UI elements in Unified Settings to the `NuGet.Configuration` APIs, we must implement an `IExternalSettingsProvider`.
+USX will invoke the following on our provider:
+- `GetValueAsync` is called by USX when obtaining values from the `NuGet.Configuration`.
+
+- `SetValueAsync` is called by USX when the user has made a change to the setting and it needs to be persisted in the backing store.
+
+The provider will respond to those API calls using the following `NuGet.Configuration` types for retrieving and persisting NuGet Configuration data:
+
+- BindingRedirectBehavior
+- PackageRestoreConsent
+- PackageManagementFormat
+
+Other notes:
+- The provider is exported by providing GUID in the `registration.json`.
+
+- The `ISettings` service will be obtained by the provider using ServiceLocator.
+
+- Any `SettingsChanged` events will invalidate the provider's state and cause USX to repopulate the page's data.
+
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+No drawbacks have been identified.
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+- The Clear Local Resources command could first open the modal dialog and provide a "start" button to allow the customer to begin the process of clearing the resources.
+This would be a change from the legacy behavior, but customers may be curious about the new USX experience, and initiate the command by mistake.
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+
+To achieve parity with the legacy Visual Studio Options for NuGet, those UI elements and implementation will serve as a starting point for the migration to Unified Settings.
+
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+
+No questions have been identified.
+
+## Future Possibilities
+
+<!-- What future possibilities can you think of that this proposal would help with? -->
+
+- `LocalsCommandRunner` could be enhanced to support Cancellation.
+This would allow the modal dialog shown from our USX command to provide a `Cancel` button in case the user decides not to wait on a large delete operation to complete and block usage of VS.
```

Comment:
Keep in mind that making delete support cancellation will significantly increase the risk that customers will experience:
* https://github.com/NuGet/Home/issues/13191

If we can make the button flush the 'fast no-op' cache, then that would be sufficient. Just something to remember if this goes ahead.

Reply:
Thanks! Added to Future Possibilities.